rasa/core/agent.py

```diff
@@ -827,6 +831,14 @@ def create_processor(
             message_preprocessor=preprocessor,
         )
 
+    def on_tracker_store_error(self, error: Exception) -> None:
```

Comment:
Could this be placed in `FailSafeTrackerStore` instead? And instead of making `FailSafeTrackerStore.__init__` receive an `on_tracker_store_error` parameter, simply allow users to subclass it and override the method with their own error callbacks.

Reply:
I think doing that will allow us to reduce the number of changes to `Agent`, and simplify `FailSafeTrackerStore` a bit.

Reply:
I prefer having the option to change the behavior on runtime. However, how about providing a default value for it?